cmake/std/atdm/ATDMDevEnvSettings.cmake

```diff
@@ -217,10 +217,6 @@ IF ((ATDM_COMPILER MATCHES ".*CLANG.*") AND ATDM_ADDRESS_SANITIZER)
   SET(EXTRA_EXTRA_LINK_FLAGS
     "${EXTRA_EXTRA_LINK_FLAGS} -ldl -fsanitize=address")
 ENDIF()
-# FPIC
-IF (ATDM_FPIC)
-  SET(CMAKE_POSITION_INDEPENDENT_CODE ON CACHE BOOL "Build targets with position independent code")
-ENDIF()
```

Comment:
@PhilMiller, I just realized that some SPARC builds specifically set this fpic option for shared builds.  Therefore, to remove this will break SPARC's usage.

How about we change the default for fpic from on to off and require those shared builds that want to set fpic to have to on have to include the keyword `fpic` the build name.  So we would upate:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#fpic

Then we could leave this logic here?

I think to change the default we just need to edit the following files:

* https://github.com/trilinos/Trilinos/edit/develop/cmake/std/atdm/README.md (starting at line 217)
* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/utils/set_build_options.sh (at line 73)

Reply:
Which does SPARC actually want, `-fPIC` so it can link objects into shared libraries, or `-fPIE` to link objects into Position-Independent Executables? PIE is generally only desirable for relocatable loading to support ASLR in a server process, or for very niche use cases in which other processes will load an executable image as a dynamic library.

The `ATDM_FPIC` setting is currently asking CMake to provide PIE, according to https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html#prop_tgt:POSITION_INDEPENDENT_CODE and its observed behavior, which besides breaking builds, also hurts performance in circumstances where it's not actually necessary.

Reply:
@rppawlo asked me to weigh in. SPARC wants Trilinos static libraries so that we can link them into either shared or static builds of our application executables.